packages/run/src/commands/logs.ts

```diff
@@ -0,0 +1,45 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {ProcessTypeCompletion} from '@heroku-cli/command/lib/completions'
+
+import logDisplayer from '../lib/log-displayer'
+
+export default class Logs extends Command {
+  static description = `display recent log output
+disable colors with --no-color, HEROKU_LOGS_COLOR=0, or HEROKU_COLOR=0`
+
+  static examples = [
+    '$ heroku logs --app=your-app',
+    '$ heroku logs --num=50',
+    '$ heroku logs --dyno=some-dyno --app=your-app',
+    '$ heroku logs --app=your-app --tail'
+  ]
+
+  static flags = {
+    app: flags.app(),
+    num: flags.integer({char: 'n', description: 'number of lines to display'}),
+    ps: flags.string({char: 'p', description: 'hidden alias for dyno', hidden: true}),
+    dyno: flags.string({
+      char: 'd',
+      description: 'only show output from this dyno type (such as "web" or "worker")',
+      completion: ProcessTypeCompletion
```

Comment:
It looks like this previously had a `DynoCompletion` and a `ProcessTypeCompletion` for the `source` flag.

Reply:
Yea, this is something I talked about with @RasPhilCo a little while ago. As near as I can tell, those completions never worked :(

There doesn't even appear to be a `DynoCompletion` in the [completions file](https://github.com/heroku/heroku-cli-command/blob/master/src/completions.ts). There *is* a `DynoSizeCompletion`, but I don't think that actually applies here since we're not passing dyno sizes as flags.

That said, I'm not sure if `ProcessTypeCompletion` applies either.

Reply:
Well then! Sounds like you've already looked into it. I'll mark this conversation as resolved.